packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +188,180 @@ export default class TraceStatistics extends Component<Props, State> {
-            name={oneSpan.name}
-            searchColor={searchColor}
-            values={values}
-            columnsArray={columnsArray}
-            togglePopup={this.togglePopup}
-            valueNameSelector1={this.state.valueNameSelector1}
-            valueNameSelector2={this.state.valueNameSelector2}
-            color={color}
-            clickColumn={this.clickColumn}
-            colorToPercent={colorToPercent}
-          />
-        );
+  render() {
+    const onClickOption = (type: string, name: string) => {
+      if (this.state.valueNameSelector1 === 'sql.query' && type !== 'undefined') this.togglePopup(name);
+    };
+
+    const sorterFunction = (a: ISorterInput, b: ISorterInput) => {
+      if (a.type === 'undefined') {
+        return 0;
       }
-      return (
-        <DetailTableData
-          key={uid}
-          type={oneSpan.type}
-          name={oneSpan.name}
-          searchColor={searchColor}
-          values={values}
-          columnsArray={columnsArray}
-          color={color}
-          togglePopup={this.togglePopup}
-          valueNameSelector2={this.state.valueNameSelector2}
-          colorToPercent={colorToPercent}
-        />
-      );
-    });
-  }
+      if (b.type === 'undefined') {
+        return -1;
+      }
+      if (typeof a.index === 'number' && typeof b.index === 'number') return a.index - b.index;
+      return (a.index as string).localeCompare(b.index as string);
+    };
 
-  renderTableHead() {
-    const { sortAsc, sortIndex } = this.state;
-    return (
-      <tr>
-        {columnsArray.map((element: any, index: number) => (
-          <HeaderTable
-            element={element}
-            key={element.title}
-            sortIndex={sortIndex}
-            index={index}
-            sortClick={this.sortClick}
-            sortAsc={sortAsc}
-          />
-        ))}
-      </tr>
-    );
-  }
+    const onCellFunction = (record: ITableSpan) => {
+      if (this.props.uiFind && record.searchColor !== 'transparent') {
+        return {
+          style: { background: record.searchColor, borderColor: record.searchColor },
+        };
+      }
+      return {
+        style: { background: record.colorToPercent, borderColor: record.colorToPercent },
+      };
+    };
 
-  render() {
+    const columns: ColumnProps<ITableSpan>[] = [
+      {
+        title: 'Name',
+        dataIndex: 'name',
+        sorter: (a, b) => sorterFunction({ type: a.type, index: a.name }, { type: b.type, index: b.name }),
+        render: (name: string, row: ITableSpan) => {
+          if (!row.color) {
+            return (
+              <span
+                onClick={() => onClickOption(row.type, row.name)}
+                style={{
+                  borderLeft: `4px solid transparent`,
+                  padding: '7px 0px 7px 10px',
+                  cursor: 'default',
+                }}
+              >
+                {name}
+              </span>
+            );
+          }
+          return (
```

Comment:
This return is nearly identical to the one above, can we merge into one and just handle color inline?